platform/viewer/src/routes/Mode/Mode.tsx

```diff
@@ -24,13 +24,20 @@ function defaultRouteInit({ servicesManager, studyInstanceUIDs, dataSource }) {
   } = servicesManager.services;
 
   const unsubscriptions = [];
-  // TODO: This should be baked into core, not manual?
-  // DisplaySetService would wire this up?
   const {
     unsubscribe: instanceAddedUnsubscribe,
   } = DicomMetadataStore.subscribe(
     DicomMetadataStore.EVENTS.INSTANCES_ADDED,
-    ({ StudyInstanceUID, SeriesInstanceUID, madeInClient = false }) => {
+    // Todo: DisplaySetService makes the displayset when ALL instances are added
```

Comment:
I don't think this is the correct description - adding all instances at a series level isn't a huge performance bottleneck unless all instances are being loaded separately.  In fact, having all instances added makes the code quite a bit faster because it allows sharing initialization/setup, and avoids using the event loop to add instances, as would be required for adding a single item first and then adding additional items. 
There is also the vary serious issue that the wrong item can be shown - images can easily come back out of order and if the display ordering is such the user expects ordering from oldest to newest or vice-versa, and gets the opposite ordering initially, then they can end up looking at things while additional items are still loading and getting it wrong.
Finally, if the hanging protocol selects various possibilities of images, and first match gets shown initially, with later ones being still loading, then the wrong series can be shown entirely - that is even worse than wrong image, as it may not be obvious.  I think suggesting anything of the sort as to multiple phases is very premature.